### DISCUSSION TITLE
Internal classes and interfaces for custom control development.

### DISCUSSION THREAD

**Discussion by richard-perry** [dotnet/maui#15932] [General]
There are several classes and interfaces that are used through the core control libraries IFontElement, FontElement, ITextElement, TextElement, ITextAlignmentElement, TextAlignmentElement to name a few.

It would be handy to have access to these for custom control development.

Was curious to what the reasoning is for these being internal or marked as non browsable.

**Comment by DanTravison** [comment]
I had the same thought. The pattern was compelling enough that I ended up creating a set of interfaces and element classes for my custom controls.

For example:
TextElement.cs
ITextElement.cs
ITextElementChanged.cs - allows TextElement to optionally invoke a property changed delegate.

It was also a perfect place to add instrumentation.

I suggest the Maui team consider making the pattern reusable.
